book/src/spec.md

```diff
@@ -0,0 +1,628 @@
+Physical stream specification
+=============================
+
+A Tydi physical stream carries a stream of elements, dimensionality information
+for said elements, and (optionally) user-defined transfer information from a
+source to a sink. The contents of these three groups and to some extent the way
+in which they are transferred are defined in a parameterized way, to provide
+the necessary flexibility for the higher levels of this specification, and to
+give hardware developers the freedom to optimize their designs. Based on these
+parameters, this specification defines which signals must exist on the
+interface, and what the significance of these signals is.
+
+Parameters
+----------
+
+A physical stream interface has five parameters, described in the following
+subsections.
+
+> Note that we define the parameters on the interfaces of the source and the
+> sink, not on the stream between them! This is because the complexity
+> parameter need not be the same on the source and sink. The others do.
+
+### Element content (E)
+
+This describes the physical layout of the elementary data being transferred by
+the stream. It is an emptyable, ordered set of named fields, where each field
+has a name and a bit count.
+
+> The element type can be empty to make a "null" stream. Such streams can still
+> be useful, as physical streams also carry metadata.
+
+The letter \\(E\\) is used as a shorthand for the total number of bits in the
+element; that is, the sum of the field bit count over all fields in the
+element.
+
+#### Field name
+
+The name of each field is a string consisting of letters, numbers, and/or
+underscores.
+
+The name cannot start or end with an underscore.
+
+The name cannot start with a digit.
+
+> It is illegal to start or end with an underscore or start with a number to
+> prevent confusion when the name is prefixed to form the signal name, and
+> for compatibility with VHDL.
+
+The name cannot be empty.
+
+> Fields may be auto-named by the next layer of the spec based on the abstract
+> type if necessary.
+
+The name must be case-insensitively unique within the set of named fields.
+
+> The identifier is case-insensitive because compatibility with VHDL is
+> desired.
+
+#### Bitcount
+
+The bit count cannot be zero.
```

Comment:
"Element type" refers to the "emptyable, ordered set of named fields, where each field
has a name and a bit count." So there wouldn't be a bit count, because there are no fields. Changed to "The element type can be given zero fields to make a 'null' stream." to clarify.

Reply:
Assuming thumbs-up means agreed, fixed in 129bbb9c